front/lib/amplitude/back/index.ts

```diff
@@ -12,10 +17,71 @@ export function getBackendClient() {
 
   const disabled = !(AMPLITUDE_ENABLED === "true");
   ampli.load({
-    environment: "dust",
+    // The environment property is a depreacted value, but still needed by the SDK. We don't use it.
+    environment: "dustprod",
+    client: {
+      apiKey: AMPLITUDE_PUBLIC_API_KEY,
+    },
     disabled: disabled,
   });
   BACKEND_CLIENT = ampli;
 
   return BACKEND_CLIENT;
 }
+
+export async function trackWorkspaceMember(membership: Membership) {
+  const amplitude = getBackendClient();
+  const user = await User.findByPk(membership.userId);
+  const workspace = await Workspace.findByPk(membership.workspaceId);
+  if (user && workspace) {
+    amplitude.client.setGroup("Group Id", workspace.sId, {
+      user_id: `user-${user.id}`,
+      time: membership.createdAt.getTime(),
+      insert_id: `membership_${user.id}-${workspace.sId}`,
+    });
+    amplitude.client.setGroup("Group Name", workspace.name, {
+      user_id: `user-${user.id}`,
+      time: membership.createdAt.getTime(),
+      insert_id: `membership_${user.id}-${workspace.name}`,
+    });
+  }
+}
+
+export async function trackUser(user: UserType) {
+  const amplitude = getBackendClient();
+  amplitude.identify(`user-${user.id}`, {});
+  amplitude.signUp(`user-${user.id}`, {
+    insert_id: `signup_${user.id}`,
+    time: user.createdAt,
+  });
+}
+
+export async function trackUserMessage({
+  userMessage,
+  workspace,
+  userId,
+  conversationId,
+}: {
+  userMessage: UserMessageType;
+  workspace: WorkspaceType;
+  userId: string;
+  conversationId: string;
+}) {
+  const amplitude = getBackendClient();
+  amplitude.userMessagePosted(
```

Comment:
Unawaited promise?

Reply:
We are not waiting for amplitude promise, they are wrapped in an object anyway, and the logger will let us know if something went wrong. This function should not be async.